File: leetcode/go/1509.minimum-difference-between-largest-and-smallest-value-in-three-moves.go
package main

import "slices"

func minDifference(nums []int) int {
	if len(nums) < 5 {
		return 0
	}
	slices.Sort(nums)

	left, right := 0, len(nums)-1
	return min(
		nums[right-3]-nums[left],
		min(
			nums[right-2]-nums[left+1],
			min(
				nums[right-1]-nums[left+2],
				nums[right]-nums[left+3],
			),
		),
	)
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

File: leetcode/rust-leetcode/src/1509.minimum-difference-between-largest-and-smallest-value-in-three-moves.rs
impl Solution {
    pub fn min_difference(mut nums: Vec<i32>) -> i32 {
        if nums.len() < 5 {
            return 0;
        }
        let mut ans = i32::MAX;
        nums.sort();
        for i in 0..4 {
            ans = ans.min(nums[nums.len() - 4 + i] - nums[i]);
        }
        ans
    }
}

